lib/core/src/Cardano/Wallet/Address/Pool.hs

```diff
@@ -193,6 +206,9 @@ usedAddresses :: Pool addr ix -> [addr]
 usedAddresses pool =
     [ addr | (addr,(_,Used)) <- Map.toList $ addresses pool ]
 
+nextIndex :: Enum ix => Pool addr ix -> ix
```

Comment:
```suggestion
-- | The first index that is 'Unused' and that comes after any 'Used' index.
nextIndex :: Enum ix => Pool addr ix -> ix
```

Added definition in comment. For an abstract data type, the idea is that it's sufficient to read the comments — no need to look at the code. 🤓